Secret-agency-web/src/main/react/secret-agency/src/App.tsx

```diff
@@ -1,8 +1,90 @@
 import * as React from 'react';
+import {MouseEvent} from 'react';
 import './App.css';
+import {LoginPage} from "./components/LoginForm";
+import * as authenticationService from "./services/authenticationService";
+import * as secretAgencyRepository from "./repository/secretAgecyRepository";
+import {IAgent} from "./types/Agent";
+
+interface IAppState {
+  loginError: string;
+  authenticatedAgent: IAgent | null;
+}
+
+type IState = IAppState;
+
+/**
+ * Main component. Shows either login form, or home page.
+ */
+class App extends React.Component<{}, IState> {
+
+  /********************************************************
+   * EVENT HANDLERS
+   *******************************************************/
+  /**
+   * Authentication method. Try to authenticate agent with given credentials.
+   * @param codeName codename
+   * @param password password
+   */
+  private onLogin = (codeName: string, password: string): void => {
+    authenticationService.authenticate(codeName, password)
+      .then((response) => {
+        // error was returned
+        if (typeof response === "string") {
+          this.setState((prevState) => ({...prevState, loginError: response}));
+        } else {
```

Comment:
tu bych mozna dal jeste check instanceof IAgent a do nove else vetve dal `unknown error `, at to mame min nachylne na errory

Reply:
OK

Reply:
jj dosel jsem pak k tomu, ze to je otypovane, tak jebat